src/Frontend/src/components/messages/HeadersView.vue

```diff
@@ -1,23 +1,91 @@
 <script setup lang="ts">
 import { ExtendedFailedMessage } from "@/resources/FailedMessage";
-
+import CopyToClipboard from "@/components/CopyToClipboard.vue";
+import { computed, ref } from "vue";
 const props = defineProps<{
   message: ExtendedFailedMessage;
 }>();
+
+const hoverStates = ref<Record<number, boolean>>({});
+const searchTerm = ref<string>("");
+const toggleHover = (index: number, state: boolean) => {
+  hoverStates.value[index] = state;
+};
+// Computed property to filter headers based on search term
+const filteredHeaders = computed(() => {
+  if (!searchTerm.value) {
+    return props.message.headers;
+  }
+  return props.message.headers.filter((header) => header.key.toLowerCase().includes(searchTerm.value.toLowerCase()) || header.value?.toLowerCase().includes(searchTerm.value.toLowerCase()));
+});
 </script>
 
 <template>
-  <table class="table" v-if="!props.message.headersNotFound">
+  <div>
+    <div class="row filters">
+      <div class="col">
+        <div class="text-search-container">
+          <div class="text-search">
+            <input type="search" aria-label="Filter by name" v-model="searchTerm" class="form-control format-text" placeholder="Search for a header key or value..." />
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+  <table class="table" v-if="filteredHeaders.length > 0 && !props.message.headersNotFound">
     <tbody>
-      <tr class="interactiveList" v-for="(header, index) in props.message.headers" :key="index">
+      <tr class="interactiveList" v-for="(header, index) in filteredHeaders" :key="index">
         <td nowrap="nowrap">{{ header.key }}</td>
         <td>
-          <pre>{{ header.value }}</pre>
+          <div class="headercopy" @mouseover="toggleHover(index, true)" @mouseleave="toggleHover(index, false)">
+            <pre>{{ header.value }}</pre>
+            <CopyToClipboard v-if="hoverStates[index]" :value="header.value || ''" :isIconOnly="true" />
+          </div>
         </td>
       </tr>
     </tbody>
   </table>
+
+  <!-- Message if filtered list is empty -->
+  <div v-if="filteredHeaders.length <= 0 && !props.message.headersNotFound" class="alert alert-warning">No headers found matching the search term.</div>
```

Comment:
There is a `NoData` component. We should use that here.

Reply:
Will do that in a separate PR